### PR TITLE
Fix camp expansion bug

### LINKED ISSUE #31141: Faction camp bug
# Describe the bug

Faction camp cannot upgrade expansions through bulletin board.

Another issue (maybe not?): Number of expansions available are dependent on the number of followers you have.

# Steps To Reproduce

1. Make a camp with any expansion
2. Interact with bulletin board, unable to tab to other expansions

# Screenshots

image

# Additional context

Save file:
[CampBug.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3254945/CampBug.zip)

### LINKED ISSUE #31147: Faction Camp can't access extend base plots
# Describe the bug
Faction Camp can't access extend base plots IE the kitchen or garage.

After extending a faction Camp to include an extra section. Kitchen, and garage tested. You get the first level of the section, ground cleared and a few pits, but are unable access their section of the interface.

# Steps To Reproduce
1. Build a faction camp
2. upgrade the camp until you can "expand base"
3. Expand base 
4. Try to view the new section

# Expected behavior

# Steps To Reproduce
1. Build a faction camp
2. upgrade the camp until you can "expand base"
3. Expand base 
4. view the new section by pressing < or >

# Screenshots

**If applicable**, add screenshots to help explain your problem.

Camp1

Camp2

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1809
- Game Version: 0.D-3949-ge061030 [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food]
]

# Additional context

[Rancocas.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3255582/Rancocas.zip)

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Bugfixes "Fix camp expansion bug"```

#### Purpose of change
Fixes #31147 
Fixes #31141 

#### Describe the solution
Add sanity check to basecamp deserialize code that checks if an expansion is present, and if so, gives it the baseline provides.

#### Describe alternatives you've considered
N/A

#### Additional context
I don't know why the "provides" of that expansion went missing in the affected saves, but this will ensure it dosnt happen again.

### CODE REVIEW

File: src/savegame_json.cpp [hunk 1]
```diff
@@ -3103,6 +3103,16 @@ void basecamp::deserialize( JsonIn &jsin )
                 e.provides[ id ] = amount;
             }
         }
+        // incase of save corruption, sanity check provides from expansions
```
Comment (Night-Pryanik) [inline src/savegame_json.cpp]:
```suggestion
        // in case of save corruption, sanity check provides from expansions
```

**Comment by davidpwbrown** [comment]
jenkins rebuld

**Comment by davidpwbrown** [comment]
jenkins rebuild

**Comment by davidpwbrown** [comment]
jenkins rebuild

**Pull request merged**